### PR TITLE
feat(zimbra): add page to add domain

### PR DESCRIPTION
**Pull request by stif59100**
| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/zimbra
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | MANAGER-14004
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] ~~Breaking change is mentioned in relevant commits~~

## Description

I added page for add domain

### CODE REVIEW

File: packages/manager/apps/zimbra/src/routes/routes.tsx [hunk 1]
```diff
@@ -18,9 +17,12 @@ const lazyRouteConfig = (importFn: CallableFunction): Partial<RouteObject> => {
 export const Routes: any = [
   {
     path: '',
-    handle: { breadcrumb: (): string => 'Zimbra' },
     ...lazyRouteConfig(() => import('@/pages/layout')),
     children: [
+      {
+        path: 'add-domain/::serviceName',
+        element: <div>toto</div>,
```
Comment (antonymarion) [inline packages/manager/apps/zimbra/src/routes/routes.tsx:24]: Toto is normal ?

File: packages/manager/apps/zimbra/src/hooks/useDomains.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import { useQuery } from '@tanstack/react-query';
+import { usePlatform } from '@/hooks';
+import { useOrganization } from './useOrganization'; // Assurez-vous de remplacer le chemin correct si nécessaire
```
Comment (antonymarion) [inline packages/manager/apps/zimbra/src/hooks/useDomains.ts:3]: in english maybe ?

File: packages/manager/apps/zimbra/src/routes/routes.tsx [hunk 3]
```diff
@@ -18,9 +17,11 @@ const lazyRouteConfig = (importFn: CallableFunction): Partial<RouteObject> => {
 export const Routes: any = [
   {
     path: '',
-    handle: { breadcrumb: (): string => 'Zimbra' },
     ...lazyRouteConfig(() => import('@/pages/layout')),
     children: [
+      {
+        path: 'add-domain/::serviceName',
+      },
```
Comment (ghyenne) [inline packages/manager/apps/zimbra/src/routes/routes.tsx:24]:
```suggestion
```

File: packages/manager/apps/zimbra/src/pages/dashboard/Domains/Domains.tsx [hunk 4]
```diff
@@ -57,13 +56,19 @@ const columns: DatagridColumn<DomainsItem>[] = [
...
   const handleAddDomain = () => {
-    console.log('Ajouter un nouveau domaine');
+    let urlWithParams = `${urls['add-domain'].replace(
+      ':serviceName',
+      platformId,
+    )}`;
+    if (organization?.id) {
+      urlWithParams += `?organizationId=${organization.id}`;
+    }
+    navigate(urlWithParams);
```
Comment (ghyenne) [inline packages/manager/apps/zimbra/src/pages/dashboard/Domains/Domains.tsx:71]: Maybe we can use useHref
```suggestion
const hrefAddDomain = useHref(`./add-domain${organization?.id ? organizationId=organization.id : '' }`);
```

File: packages/manager/apps/zimbra/src/pages/dashboard/Domains/Domains.tsx [hunk 5]
```diff
@@ -57,13 +56,19 @@ const columns: DatagridColumn<DomainsItem>[] = [
...
   const { t } = useTranslation('domains');
   const { platformId } = usePlatform();
+  const { data } = useDomains();
+  const navigate = useNavigate();
   const { data: organization } = useOrganization();
-  const { data } = useQuery({
-    queryKey: getZimbraPlatformDomainsQueryKey(platformId, organization?.id),
-    queryFn: () => getZimbraPlatformDomains(platformId, organization?.id),
-  });
+  const shouldHideContent = useOverridePage();
```
Comment (ghyenne) [inline packages/manager/apps/zimbra/src/pages/dashboard/Domains/Domains.tsx:62]:
```suggestion
  const isOverriddedPage = useOverridePage();
```

File: packages/manager/apps/zimbra/src/pages/dashboard/Domains/Domains.tsx [hunk 6]
```diff
@@ -72,34 +77,40 @@ export default function Domains() {
...
-          inline={true}
-          size={ODS_BUTTON_SIZE.sm}
-          onClick={handleAddDomain}
-        >
-          <span slot="start">
-            <OsdsIcon
-              name={ODS_ICON_NAME.PLUS}
-              size={ODS_ICON_SIZE.sm}
+    <>
+      {!shouldHideContent && (
```
Comment (ghyenne) [inline packages/manager/apps/zimbra/src/pages/dashboard/Domains/Domains.tsx:83]:
```suggestion
      {!isOverriddedPage && (
```

File: packages/manager/apps/zimbra/src/pages/dashboard/Domains/Domains.tsx [hunk 7]
```diff
@@ -72,34 +77,40 @@ export default function Domains() {
...
-          ...column,
-          label: t(column.label),
-        }))}
-        items={items}
-        totalItems={items.length}
-      />
-    </div>
+              inline={true}
+              size={ODS_BUTTON_SIZE.sm}
+              onClick={handleAddDomain}
```
Comment (ghyenne) [inline packages/manager/apps/zimbra/src/pages/dashboard/Domains/Domains.tsx:90]:
```suggestion
              href={hrefAddDomain}
```

**Review by ghyenne — CHANGES_REQUESTED** [review]

**Review by ghyenne — CHANGES_REQUESTED** [review]

File: packages/manager/apps/zimbra/src/pages/dashboard/Domains/AddDomain.tsx [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+            <div slot="helper">
+              <OsdsSpinner
+                color={ODS_THEME_COLOR_INTENT.primary}
+                size={ODS_SPINNER_SIZE.sm}
+                inline
+              />
+            </div>
+          ) : (
+            ''
+          )}
```
Comment (ghyenne) [inline packages/manager/apps/zimbra/src/pages/dashboard/Domains/AddDomain.tsx:75]:
```suggestion
          {isLoading && (
            <div slot="helper">
              <OsdsSpinner
                color={ODS_THEME_COLOR_INTENT.primary}
                size={ODS_SPINNER_SIZE.sm}
                inline
              />
            </div>
          )}
```

**Review by ghyenne — CHANGES_REQUESTED** [review]

**Review by ghyenne — APPROVED** [review]

**Review by antonymarion — APPROVED** [review]

**Pull request merged**